### PR TITLE
SW-586_disable_non_available_filters

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

File: tests/Storefront/Controller/SearchControllerTest.php [hunk 1]
```diff
@@ -81,4 +81,76 @@ public function testAvailableFilterReturnsCorrectResponse(string $demoResponse,
...
+
+    public function testFiltersWhichAreNotInTheAvailleFilterResponseAreStillReturned(): void
+    {
+        if (Utils::versionLowerThan('6.3.3.0')) {
+            $this->markTestSkipped('Filter disabling feature was introduced in Shopware 6.3.3.0');
+        }
+
+        $allFiltersResponse = new Xml21Response($this->getMockResponse('XMLResponse/demoResponseWithNoResults.xml'));
+        $responseParser = new Xml21ResponseParser($allFiltersResponse);
+        $availableFilters = $responseParser->getFiltersExtension();
```
Comment (dblaichinger) [inline tests/Storefront/Controller/SearchControllerTest.php:93]: `$allFiltersResponse` is used for `$availableFilters`? Please try to improve naming

File: tests/Storefront/Controller/SearchControllerTest.php [hunk 2]
```diff
@@ -81,4 +81,76 @@ public function testAvailableFilterReturnsCorrectResponse(string $demoResponse,
...
+
+        $response = new Xml21Response($this->getMockResponse('XMLResponse/demoResponseWithAvailableFilters.xml'));
```
Comment (dblaichinger) [inline tests/Storefront/Controller/SearchControllerTest.php:95]: `demoResponseWithAvailableFilters.xml` is used for `$allFilters`? Please try to improve naming.

File: tests/Storefront/Controller/SearchControllerTest.php [hunk 3]
```diff
@@ -81,4 +81,76 @@ public function testAvailableFilterReturnsCorrectResponse(string $demoResponse,
...
+        $eventMock->method('getRequest')->willReturn($request);
+        $eventMock->method('getCriteria')->willReturn($criteria);
+
+        $filterHandler = new FilterHandler();
+        $filterResponse = $filterHandler->handleAvailableFilters($eventMock);
+        $expectedFilters = [
+            'properties' => [
+                'entities' => [
+                    'rating' => [
+                        'max' => 0
```
Comment (dblaichinger) [inline tests/Storefront/Controller/SearchControllerTest.php:115]: This means only rating is available, others are disabled? It's probably only possible to see that when checking both demo xml responses, right?

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: src/Findologic/Request/Handler/FilterHandler.php [hunk 4]
```diff
@@ -217,19 +217,26 @@ private function handlePropertyFilter(
...
+        FiltersExtension $allFilters
+    ): array {
         $result = [];
         $result[RatingFilter::RATING_FILTER_NAME]['max'] = 0;
 
-        foreach ($filterExtension->getFilters() as $filter) {
-            $filterName = $filter->getId();
+        foreach ($allFilters->getFilters() as $filterWithAllValues) {
+            $filterName = $filterWithAllValues->getId();
+            if (!$filter = $availableFilters->getFilter($filterName)) {
```
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Findologic/Request/Handler/FilterHandler.php:236]: Of course. Just a few lines below:

https://github.com/findologic/plugin-shopware-6/blob/c1a3e3cc48156f22272edde0f0f2331b0ba92f77/src/Findologic/Request/Handler/FilterHandler.php#L242

https://github.com/findologic/plugin-shopware-6/blob/c1a3e3cc48156f22272edde0f0f2331b0ba92f77/src/Findologic/Request/Handler/FilterHandler.php#L244

File: tests/Storefront/Controller/SearchControllerTest.php [hunk 5]
```diff
@@ -71,7 +71,7 @@ public function testAvailableFilterReturnsCorrectResponse(string $demoResponse,
 
         $request = new Request();
         $criteria = new Criteria();
-        $criteria->setExtensions(['flAvailableFilters' => $filterExtension]);
+        $criteria->setExtensions(['flAvailableFilters' => $filterExtension, 'flFilters' => $filterExtension]);
```
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply tests/Storefront/Controller/SearchControllerTest.php:74]: This test is ensuring that the structure is properly set. We need `flFilters`, so we know which filters to even return.

File: tests/Storefront/Controller/SearchControllerTest.php [hunk 6]
```diff
@@ -81,4 +81,76 @@ public function testAvailableFilterReturnsCorrectResponse(string $demoResponse,
 
         $this->assertSame($filterResponse, $expectedFilters);
     }
+
+    public function testFiltersWhichAreNotInTheAvailleFilterResponseAreStillReturned(): void
```
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply tests/Storefront/Controller/SearchControllerTest.php:85]: Oops :smile:

File: tests/Storefront/Controller/SearchControllerTest.php (hunk 3 above)
Comment (TheKeymaster) [reply tests/Storefront/Controller/SearchControllerTest.php:115]: Also rating is disabled. What I could see from Huzaifas commit messages, is, that we need this in order for Shopware to disable the Rating filter properly.

File: tests/Storefront/Controller/SearchControllerTest.php (hunk 2 above)
Comment (TheKeymaster) [reply tests/Storefront/Controller/SearchControllerTest.php:95]: This response is used by other tests as well. Now using `demoResponseWithAllFilterTypes.xml` instead, to make this clearer.

File: src/Findologic/Request/Handler/FilterHandler.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (dblaichinger) [reply src/Findologic/Request/Handler/FilterHandler.php:236]: Right, didn't saw that.

**Review by dblaichinger — APPROVED** [review]

**Pull request merged**